legal-api/src/legal_api/services/authz.py

```diff
@@ -95,96 +107,441 @@ def has_roles(jwt: JwtManager, roles: List[str]) -> bool:
-                'limitedRestorationToFull': ['BC', 'BEN', 'CC', 'ULC', 'LLC']
+                'limitedRestorationExtension': {
+                    'legalTypes': ['BC', 'BEN', 'CC', 'ULC', 'LLC'],
+                    'blockerChecks': {
+                        'validStateFilings': ['restoration.limitedRestoration',
+                                              'restoration.limitedRestorationExtension'],
+                        'business': [BusinessBlocker.DEFAULT]
+                    }
+                },
+                'limitedRestorationToFull': {
+                    'legalTypes': ['BC', 'BEN', 'CC', 'ULC', 'LLC'],
+                    'blockerChecks': {
+                        'validStateFilings': ['restoration.limitedRestoration',
+                                              'restoration.limitedRestorationExtension'],
+                        'business': [BusinessBlocker.DEFAULT]
+                    }
+                }
             }
         },
         Business.State.HISTORICAL: {
-            'courtOrder': ['SP', 'GP', 'CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC'],
-            'putBackOn': ['SP', 'GP', 'BEN', 'CP', 'BC', 'CC', 'ULC'],
-            'registrarsNotation': ['SP', 'GP', 'CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC'],
-            'registrarsOrder': ['SP', 'GP', 'CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC'],
+            'courtOrder': {
+                'legalTypes': ['SP', 'GP', 'CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC'],
+            },
+            'putBackOn': {
+                'legalTypes': ['SP', 'GP', 'BEN', 'CP', 'BC', 'CC', 'ULC'],
+                'blockerChecks': {
+                    # FUTURE add outstanding filings to invalidStateFilings when implemented
+                    'invalidStateFilings': []
+                }
+            },
+            'registrarsNotation': {
+                'legalTypes': ['SP', 'GP', 'CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC']
+            },
+            'registrarsOrder': {
+                'legalTypes': ['SP', 'GP', 'CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC']
+            },
             'restoration': {
-                'fullRestoration': ['CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC'],
-                'limitedRestoration': ['BC', 'BEN', 'CC', 'ULC', 'LLC']
+                'fullRestoration': {
+                    'legalTypes': ['CP', 'BC', 'BEN', 'CC', 'ULC'],
+                    'blockerChecks': {
+                        # FUTURE add outstanding filings to invalidStateFilings when implemented
+                        'invalidStateFilings': ['continuationIn', 'continuationOut']
+                    }
+                },
+                'limitedRestoration': {
+                    'legalTypes': ['BC', 'BEN', 'CC', 'ULC'],
+                    'blockerChecks': {
+                        # FUTURE add outstanding filings to invalidStateFilings when implemented
+                        'invalidStateFilings': ['continuationIn', 'continuationOut']
+                    }
+                }
```

Comment:
`invalidStateFilings ` will only allow the target(`fullRestoration`, `limitedRestoration`) filing if the current business state filing type is not one of the filing types provided.